### DISCUSSION TITLE
Improve the M2M GraphQL queries

### DISCUSSION THREAD

**Discussion by abdonrd** [directus/directus#11917] [Feature Requests]
Right now, you need to make a GrahQL query like this for the M2M relationships:

```graphql
query MyQuery {
  matches {
    id
    ...
    participants {
      profiles_id {
        name
      }
    }
  }
}
```

```json
{
  "data": {
    "matches": {
      "id": 1,
      "participants": {
        "profiles_id": {
          "name": "Abdón"
        }
      }
    }
  }
}
```

Note the `profiles_id` from the junction collection.

I want to propose to make it easier:

```graphql
query MyQuery {
  matches {
    id
    ...
    participants {
      name
    }
  }
}
```

```json
{
  "data": {
    "matches": {
      "id": 1,
      "participants": {
        "name": "Abdón"
      }
    }
  }
}
```

**Comment by rijkvanzanten** [comment]
This is because there's no difference between junction collections and "regular" collections. A junction collection can have any number of custom fields as well. 

As of right now, a M2M is functionally no different from a O2M and a M2O

**Reply by abdonrd** [reply]
Yeah, I propose it as a feature request.
I think it would be a better experience for a very common use case.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
